File: q1.py
class Relevance:
    def __init__(self):
        with open("input.txt") as f:
            data = f.readlines()
        pages = [page for page in data if page.split(" ")[0] == 'P']
        queries = [query for query in data if query.split(" ")[0] == 'Q']
        self.pages = []
        for p in pages:
            page = [x.replace("\n", "") for x in p.split(" ")[1:]]
            self.pages.append(page)
        self.queries = []
        for q in queries:
            query = [x.replace("\n", "") for x in q.split(" ")[1:]]
            self.queries.append(query)

    def relation_query_page(self):
        self.relation = []
        for query in self.queries:
            n_page = []
            for page in self.pages:
                relation = 0
                for q in query:
                    for p in page:
                        if q == p:
                            relation = relation + (8-query.index(q)) * (8-page.index(p))

                n_page.append({
                    f"P{self.pages.index(page)+1}": relation
                })
            self.relation.append({
                f"Q{self.queries.index(query)+1}:": n_page
            })

    def outputData(self):
        print(self.relation)

File: q1.rb
class Relevance
  def initialize
    data = File.readlines("input.txt")
    pages = []
    queries = []
    for line in data
      if line.strip.split(" ")[0] == 'P'
        pages.push(line)
      else
        queries.push(line)
      end
    end
    @pages = pages
    @queries = queries
  end

  def relation_query_page
    @relation = []
    for query in @queries
      n_page = []
      for page in @pages
        relation = 0
        for q in query.split(" ")
          for p in page.split(" ")
            if q == p
              relation += (8 - query.split(" ")[1..-1].index(q)) * (8 - page.split(" ")[1..-1].index(p))
            end
          end
        end
        n_page.push({
          "P#{@pages.index(page)+1}" => relation
        })
      end
      @relation.push({"Q#{@queries.index(query)+1}" => n_page})
    end
  end

  def outputData
    print @relation
  end
end
